packages/data-point/docs/best-practices.md

```diff
@@ -0,0 +1,457 @@
+
+**Example:**
+
+Using `Accumulator.create` factory for mocking the **Accumulator** object:
+
+```js
+const Accumulator = require('data-point/lib/accumulator')
+
+function getLocalizedMonth(valuePath) {
+  return (input, acc) => {
+    const rawDate = get(input, valuePath)
+    const date = new Date(rawDate)
+    return acc.locals.i18n.months[date.getMonth()]
+  }
+}
+
+describe('getLocalizedMonth', () => {
+  it('should return localized month name', () => {
+    const acc = Accumulator.create()
+    const i18n = {
+      months: ['Enero', 'Febrero', /* rest of months... */ ]
+    }
+    acc.locals.i18n = i18n
+
+    expect(getLocalizedMonth(new Date('2018-02-04T03:24:00'))).toEqual('Febrero')
+  })
+})
+```
+
+#### Mocking a Request
+
+Internally on DataPoint, we are using [nock](https://www.npmjs.com/package/nock) to mock Request entities.
+
+Example ([examples/request-entity-mock.test.js](request-entity-mock.test.js)):
+
+```js
+const DataPoint = require('data-point')
+const nock = require('nock')
+
+const PersonByIdRequest = DataPoint.Request('PersonByIdRequest', {
+  url: 'https://swapi.co/api/people/{value}/'
+})
+
+describe('PersonByIdRequest', () => {
+  it('should request Swapi API to fetch a person by its id', async () => {
+    const dataPoint = DataPoint.create()
+
+    const response = {
+      name: 'Obi-Wan Kenobi'
+    }
+
+    nock('https://swapi.co')
+      .get('/api/people/10/')
+      .reply(200, response)
+
+    const result = await dataPoint.resolve(PersonByIdRequest, 10)
+
+    expect(result).toEqual(response)
+  })
+})
+```
+
+## Asynchronous execution
+
+By design DataPoint transformations are asynchronous, this provides maximum flexibility for writing your code but it also means that there are some things we must take into account and some patterns to be avoided to provide the best performance. It is important to know that to resolve all asynchronous processes DataPoint relies heavily on Promises([Bluebird](http://bluebirdjs.com)), take this into account when optimizing your code and how bluebird handles its promises.
+
+Patterns related to this topic:
+
+- grouping multiple synchronous reducers together
+- picking/omitting keys from an object
+- using functions to manipulate objects
+- optimizing operations on collections
+
+## Patterns
+
+### Pure Functions
+
+We should think of DataPoint only as a transformation tool, your transformations should(when possible) follow the same principles of [pure functions](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976):
+
+- Given the same inputs, always returns the same output.
+- [Referential transparency](https://medium.com/@olxc/referential-transparency-93352c2dd713) - if the expression can be replaced with its value without changing the program’s behavior.
+- Produce no side effects - the expression should not alter any external state (this includes not mutating any of the parameter values).
```

Comment:
I think this should stress that you shouldn't modify the accumulator (because it's very tempting to modify it directly). Since everything is async, modifying the accumulator could have unpredictable results if multiple reducers have a reference to the same object

Reply:
done! thanks!